cni/deployments/kubernetes/install/scripts/install-cni.sh

```diff
@@ -25,208 +25,297 @@ set -u -e
 # Helper function for raising errors
 # Usage:
 # some_command || exit_with_error "some_command_failed: maybe try..."
-exit_with_error(){
+function exit_with_error() {
   echo "$1"
   exit 1
 }
 
-exit_graceful(){
+function exit_graceful() {
   exit 0
 }
 
+# Adds the given file to FILE_CLEANUP_ARR for cleanup on exit
+function add_file_to_cleanup() {
+  FILE_CLEANUP_ARR+=( "$1" )
+}
+
 function rm_bin_files() {
   echo "Removing existing binaries"
   rm -f /host/opt/cni/bin/istio-cni /host/opt/cni/bin/istio-iptables
 }
+
+# Creates a temp file with the given filepath prefix
+# Usage:
+# TMP_FILE=$(create_temp_file "filepath/prefix")
+# Recommended usage:
+# TMP_FILE=$(create_temp_file "filepath/prefix") && add_file_to_cleanup "${TMP_FILE}"
+function create_temp_file() {
+  local filepath=$1
+  local temp_file=$(mktemp ${filepath}.tmp.XXXXXX)
+  if [ $? -ne 0 ]; then
+    echo "Failed to create temp file. Exiting (1)..."
+    exit 1
+  fi
+  # mktemp creates files with read and write permissions only for owner
+  # go tests require additional read permissions so default to 644
+  chmod 644 "${temp_file}"
+  echo "${temp_file}"
```

Comment:
Remove this debug log.

Reply:
The echo is the output for the function so the caller can store the path of the temp file in a variable since the name is randomly generated.